src/main/scala/com/cognite/spark/datasource/DataPointsRelation.scala

```diff
@@ -371,7 +155,11 @@ class DataPointsRelation(
       .map(r => {
         if (config.collectMetrics) {
           val numPoints = data.namedTimeseriesData
-            .map(_.getNumericData.points.length)
+            .map(
+              ts =>
+                ts.data.numericData
+                  .map(_.points.length)
```

Comment:
Can the flatMap at the end of this method be removed?

Reply:
It can! Well spotted, thanks.